sphinx/screens-detail/payments/payment-template/payment-template/src/main/java/chat/sphinx/payment_template/ui/PaymentTemplateFragment.kt

```diff
@@ -0,0 +1,206 @@
+            )
+
+            layoutManager = LinearLayoutManager(context, RecyclerView.HORIZONTAL, false)
+            adapter = paymentTemplateAdapter
+
+            val snapHelper: SnapHelper = LinearSnapHelper()
+            snapHelper.attachToRecyclerView(this)
+
+            addOnScrollListener(object : RecyclerView.OnScrollListener() {
+                override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
+                    super.onScrolled(recyclerView, dx, dy)
+
+                    snapHelper.findSnapView(layoutManager)?.let { centerView ->
+                        layoutManager?.getPosition(centerView)?.let { position ->
+                            if (snapPosition != position) {
+                                //Subtracting padding item and no template item from position
+                                viewModel.selectTemplate(position - 2)
+                            }
+                        }
+                    }
+                }
+            })
+        }
+
+        onStopSupervisor.scope.launch(viewModel.mainImmediate) {
+            viewModel.loadTemplateImages()
+        }
+    }
+
+    private fun setupFragmentLayout() {
+        (requireActivity() as InsetterActivity).addNavigationBarPadding(
+            binding.root
+        )
+    }
+
+    override suspend fun onSideEffectCollect(sideEffect: PaymentTemplateSideEffect) {
+        sideEffect.execute(requireActivity())
+    }
+
+    override suspend fun onViewStateFlowCollect(viewState: PaymentTemplateViewState) {
+        when (viewState) {
+            is PaymentTemplateViewState.Idle -> {}
+
+            is PaymentTemplateViewState.ProcessingPayment -> {
+                binding.progressBarConfirm.visible
+            }
+            is PaymentTemplateViewState.PaymentFailed -> {
+                binding.progressBarConfirm.gone
+            }
+        }
+    }
+
+    override fun subscribeToViewStateFlow() {
+        super.subscribeToViewStateFlow()
+
+        onStopSupervisor.scope.launch(viewModel.mainImmediate) {
+            viewModel.templateImagesViewStateContainer.collect { viewState ->
+                @Exhaustive
+                when (viewState) {
+                    is TemplateImagesViewState.LoadingTemplateImages -> {
+                        binding.layoutConstraintProgressBarLoadingTemplates.visible
+                    }
+                    is TemplateImagesViewState.TemplateImages -> {
+                        binding.layoutConstraintProgressBarLoadingTemplates.gone
+
+                        binding.recyclerViewPaymentTemplate.apply {
+
+                            (adapter as PaymentTemplateAdapter).setItems(viewState.templates)
+
+                            //scrolling to first template item
+                            scrollBy(
+                                context.resources.getDimension(R.dimen.payment_template_recycler_view_item_width).toInt(),
+                                0
+                            )
```

Comment:
upon moving application to background and then foreground again, this will proc re-collection of ViewState and scroll to 0th position + set items again unnecessarily. use local variable to track last state observed that gets set back to null upon onDestroyView lifecycle event.